ebms-provider/src/main/kotlin/no/nav/emottak/ebms/App.kt

```diff
@@ -78,20 +106,52 @@ fun main() = SuspendApp {
         }
 }
 
-fun Application.ebmsProviderModule(
+private fun CoroutineScope.launchPayloadReceiver(
+    config: Config,
     ebmsMessageDetailsRepository: EbmsMessageDetailsRepository,
-    payloadRepository: PayloadRepository
+    eventsRepository: EventsRepository,
+    dokumentValidator: DokumentValidator,
+    processingService: ProcessingService,
+    ebmsSignalProducer: EbmsSignalProducer,
+    smtpTransportClient: SmtpTransportClient
 ) {
-    val config = config()
-
-    val ebmsSignalProducer = EbmsSignalProducer(config.kafkaSignalProducer.topic, config.kafka)
-
-    val cpaClient = CpaRepoClient(defaultHttpClient())
-    val validator = DokumentValidator(cpaClient)
+    if (config.kafkaPayloadReceiver.active) {
+        launch(Dispatchers.IO) {
+            val payloadMessageProcessor = PayloadMessageProcessor(
+                ebmsMessageDetailsRepository = ebmsMessageDetailsRepository,
+                eventsRepository = eventsRepository,
+                validator = dokumentValidator,
+                processingService = processingService,
+                ebmsSignalProducer = ebmsSignalProducer,
+                smtpTransportClient = smtpTransportClient
+            )
+            startPayloadReceiver(config.kafkaPayloadReceiver.topic, config.kafka, payloadMessageProcessor)
+        }
+    }
+}
```

Comment:
Er det i hovedsak på grunn av dependency injectionss at vi fortsetter å legge til mange forskjellige klasser her? 
Hvis så, kanskje vi burde vurdere å se på DI-rammeverk. For eksempel kan vi bruke [Koin](https://insert-koin.io/), som er et populært DI-rammeverk. Jeg har ikke tatt dette i bruk selv, men kanskje @kfh har. 

Tenker dette sikkert kan hjelpe oss med å strukturere og administrere avhengigheter på en finere måte. 
Hva tenker du? Kanskje dette er noe som kan være verdt å ta opp i felleskap?